### ISSUE TITLE
Training TabNet on Higgs example fails to converge, AUC stuck at 0.5.

### ISSUE THREAD

**Issue by dantreiman** [ludwig-ai/ludwig#1677]
Training TabNet on the higgs dataset `examples/tabnet/higgs/train_higgs_small.py` does not converge.

Train loss decreases somewhat, but eval accuracy does not improve and AUC is stuck at 0.5.  No improvement observed after 87 epochs.

Run log attached:
[train_higgs_small.log](https://github.com/ludwig-ai/ludwig/files/7873562/train_higgs_small.log)

**Comment by amholler** [comment]
ForestCover has a similar issue and is a smaller dataset for debugging.

(see https://docs.google.com/spreadsheets/d/1c1ghzlNBGH8Sh0AgxZw8Z366NZegiblxOuwQB9eHHf4/edit#gid=665329770)

**Comment by amholler** [comment]
OttoGroupProduct and PokerHand seem to have similar issues.

(see https://docs.google.com/spreadsheets/d/1c1ghzlNBGH8Sh0AgxZw8Z366NZegiblxOuwQB9eHHf4/edit#gid=2057342027)

**Comment by amholler** [comment]
For OttoGroupProduct and ForestCover, the best model is produced by epoch 0; for PokerHand, it is epoch 1.
This seems very odd and hopefully is a good clue.

**Comment by dantreiman** [comment]
That was a good clue, thanks!  I think this can be fixed by removing
```
  regularization_lambda: 1,
```
from the config.

In older versions, sparsity loss was treated as a regularization loss, so this had to be set to enable sparsity.  Now sparsity loss is automatically added to the overall loss, and `regularization_type` defaults to l2.  The strong l2 regularization is overpowering the train loss.

This fixed TabNet on Higgs:  https://github.com/ludwig-ai/ludwig/pull/1682

I haven't tried the other experiments yet.

**Comment by amholler** [comment]
Interesting, thanks!  Yes, that value is currently included in automl via
ludwig/automl/defaults/combiner/tabnet_config.yaml
Let me retry w/o it for one of the datasets mentioned above.

**Comment by amholler** [comment]
Yes, I just tried forest_cover with this change and it gets the expected performance.
Will create PR and validate on other data sets.

**Comment by dantreiman** [comment]
Confirmed by Piero:
"
So the lambda was needed before because the tabnet internal sparsity loss was treated as a regularization loss and summed up with the lambda multiplier. We changed that codepath and mow each loss is summed up independently and lambda is only for l1/l2 regularization. Before the default reg was set to None and turnng on lambda in tabnet meant turning on only the tabnet loss. Now the tabnet loss is not dependent on lambda (as it should be) and the default reg il l2, so setting lambda to 0 achieves the same loss calculation as before.
"
